src/nunit.analyzers/WithinUsage/WithinUsageAnalyzer.cs

```diff
@@ -0,0 +1,122 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using NUnit.Analyzers.Constants;
+using NUnit.Analyzers.Extensions;
+using NUnit.Analyzers.Helpers;
+
+namespace NUnit.Analyzers.WithinUsage
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class WithinUsageAnalyzer : BaseAssertionAnalyzer
+    {
+        private static readonly string[] SupportedIsMethods = new[]
+        {
+            NunitFrameworkConstants.NameOfIsEqualTo
```

Comment:
```suggestion
             NunitFrameworkConstants.NameOfIsEqualTo,
            NunitFrameworkConstants.NameOfIsLessThan,
            NunitFrameworkConstants.NameOfIsLessThanOrEqualTo,
            NunitFrameworkConstants.NameOfIsGreaterThan,
            NunitFrameworkConstants.NameOfIsGreaterThanOrEqualTo,
```

You can actually do: `Assert.That(Math.PI, Is.LessThan(Math.E).Within(0.5));`